examples/smooth.py

```diff
@@ -238,64 +278,71 @@ def plot_raw_and_smooth_timeseries_and_psd(
 
 # %%
 plot_raw_and_smooth_timeseries_and_psd(
-    ds_wasp,
-    ds_wasp_savgol,
-    keypoint="stinger",
+    ds_wasp, ds_wasp_smooth, keypoint="stinger"
 )
 # %%
 # This example shows two important limitations of the Savitzky-Golay filter.
 # First, the filter can introduce artefacts around sharp boundaries. For
 # example, focus on what happens around the sudden drop in position
 # during the final second. Second, the PSD appears to have large periodic
 # drops at certain frequencies. Both of these effects vary with the
-# choice of ``window_length`` and ``polyorder``. You can read more about these
+# choice of ``window`` and ``polyorder``. You can read more about these
 # and other limitations of the Savitzky-Golay filter in
 # `this paper <https://pubs.acs.org/doi/10.1021/acsmeasuresciau.1c00054>`_.
 
 
 # %%
 # Combining multiple smoothing filters
 # ------------------------------------
-# You can also combine multiple smoothing filters by applying them
+# We can also combine multiple smoothing filters by applying them
 # sequentially. For example, we can first apply the median filter with a small
-# ``window_length`` to remove "spikes" and then apply the Savitzky-Golay filter
-# with a larger ``window_length`` to further smooth the data.
+# ``window`` to remove "spikes" and then apply the Savitzky-Golay filter
+# with a larger ``window`` to further smooth the data.
 # Between the two filters, we can interpolate over small gaps to avoid the
 # excessive proliferation of NaN values. Let's try this on the mouse dataset.
-# First, let's apply the median filter.
+# First, we will apply the median filter.
 
-ds_mouse_medfilt = median_filter(ds_mouse, window_length=0.1, min_periods=2)
+window = int(0.1 * ds_mouse.fps)
+ds_mouse_smooth.update(
+    {"position": ds_mouse.move.median_filter(window, min_periods=2)}
+)
 
 # %%
-# Next, let's linearly interpolate over gaps smaller than 1 second.
+# Next, let's linearly interpolate over gaps smaller than 1 second (30 frames).
 
-ds_mouse_medfilt_interp = interpolate_over_time(ds_mouse_medfilt, max_gap=1)
+ds_mouse_smooth.update(
+    {"position": ds_mouse_smooth.move.interpolate_over_time(max_gap=30)}
+)
 
 # %%
-# Finally, let's apply the Savitzky-Golay filter.
+# Finally, let's apply the Savitzky-Golay filter over a 0.4-second window
+# (12 frames).
 
-ds_mouse_medfilt_interp_savgol = savgol_filter(
-    ds_mouse_medfilt_interp, window_length=0.4, polyorder=2
+window = int(0.4 * ds_mouse.fps)
+ds_mouse_smooth.update(
+    {"position": ds_mouse_smooth.move.savgol_filter(window)}
 )
 
 # %%
-# A record of all applied operations is stored in the dataset's ``log``
-# attribute. Let's inspect it to summarise what we've done.
+# A record of all applied operations is stored in the ``log`` attribute of the
+# `ds_mouse_smooth.position`` data array. Let's inspect it to summarise
+# what we've done.
 
-for entry in ds_mouse_medfilt_interp_savgol.log:
+for entry in ds_mouse_smooth.position.log:
```

Comment:
Curiously, the log here appears complete. So updating the dataset preserves all entries, but assigning the array to a new variable somehow doesn't?

Reply:
Based on your response to the other comment about this issue, I understand it now.